### PR TITLE
Test simple inline modules with symbol tables.

### PR DESCRIPTION
**Pull request by toddjonker**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: conformance/ion_encoding/module/symtab.ion [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+// Retention
+
+(ion_1_1
+  (toplevel $ion_encoding::((load A "abcs" 1))
+            $ion_encoding::((retain A)
+                            (module I (symbol_table ["i"]))
+                            (symbol_table A))
+            '#$12'
+            $ion_encoding::((retain I A A)
```
Comment (popematt) [inline conformance/ion_encoding/module/symtab.ion:51]: Typo?
```suggestion
            $ion_encoding::((retain I A)
```

File: conformance/ion_encoding/module/symtab.ion [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        (encoding (module M (symbol_table ())))
+        (encoding (module M (symbol_table ("a" "b"))))
+        (encoding (module M (symbol_table ["a"] ("b"))))
+        (signals "Unexpected element in `symbol_table`")))
+
+
+// Retention
+
+(ion_1_1
+  (toplevel $ion_encoding::((load A "abcs" 1))
```
Comment (popematt) [inline conformance/ion_encoding/module/symtab.ion:46]: Any particular reason you're using `encoding` for the encoding directives in first part of the file and `toplevel` here?

File: conformance/ion_encoding/module/symtab.ion (hunk 1 above)
Comment (toddjonker) [reply conformance/ion_encoding/module/symtab.ion:51]: It tests that `retain` can contain a repeated name, but that's better suited for the related tests in https://github.com/amazon-ion/ion-tests/pull/112 so I'll move it there.

File: conformance/ion_encoding/module/symtab.ion (hunk 2 above)
Comment (toddjonker) [reply conformance/ion_encoding/module/symtab.ion:46]: Nope.  Some of these tests predate the `encoding` form and I've not prioritized updating everything.

**Review by popematt — APPROVED** [review]

**Comment by toddjonker** [comment]
Sorry @popematt we raced and I added some more changes.

**Review by popematt — APPROVED** [review]

**Pull request merged**